tartufo/config.py

```diff
@@ -243,3 +243,36 @@ def compile_path_rules(patterns: Iterable[str]) -> List[Pattern]:
         for pattern in stripped
         if pattern and not pattern.startswith("#")
     ]
+
+
+def compile_rule(pattern: str) -> Rule:
+    """
+    Compile pattern string to Rule.
+
+    :param pattern: Rule pattern with {path_pattern}::{pattern}
+    :return Rule: Rule object with pattern and path_pattern
+    """
+    parts = pattern.split("::")
+    if len(parts) == 1:
+        parts = [".*", parts[0]]
+    return Rule(
+        name=None,
+        pattern=re.compile("::".join(parts[1:])),
+        path_pattern=re.compile(parts[0]),
+    )
```

Comment:
One way to make this a little easier, instead of having to deal with `parts[0]` and `parts[1:]`, you can actually use a combination of multi-assignment and limiting the split. So you could end up with something like:

```python
try:
    path, pattern = pattern.split("::", 1)
except ValueError:  # Raised when the split separator is not found
    path = ".*"
return Rule(
    name=None,
    pattern=re.compile(pattern)
    path_pattern=re.compile(path)
)
```
I wouldn't call this a hard blocker for the PR, but figured I'd share in case you hadn't seen these features before 😄

Reply:
Thanks for the split limit. Didn't know about that. Updated.

Reply:
cc @tarkatronic